src/FormElement/ScheduleElement/WeeklyFields.php

```diff
@@ -119,8 +142,10 @@ protected function assemble()
             $listItems->addHtml($listItem);
         }
 
-        if (! $foundCheckedDay) {
-            $this->getElement($this->default)->setChecked(true);
+        if (! $foundCheckedDay && $this->getPopulatedValue($this->default) !== 'n') {
+            $this->getElement($this->default)
+                ->setValue('y')
+                ->setChecked(true);
```

Comment:
This is not required if you correctly define the initial value of each checkbox.

Reply:
I enable the checkbox after all elements are registered/added, how would the initial value help here. If then this would have to be fixed in `CheckboxElement`, so that `setChecked()` call also sets the value to `y`.